extensions/src/main/java/org/pentaho/platform/plugin/action/mondrian/catalog/MondrianCatalogHelper.java

```diff
@@ -1194,11 +1196,19 @@ org.w3c.dom.Document getMondrianXmlDocument( InputStream is )
     return builder.parse( is );
   }
 
-  private synchronized void loadCatalogIntoCache( String catalogName, final IPentahoSession pentahoSession ) {
 
+  /**
+   * Load catalog into cache. If not found or if overwrite is true, load the catalog into cache.
+   *
+   * @param catalogName    the catalog name
+   * @param pentahoSession the pentaho session where the cache is stored
+   * @param overwrite      flag to indicate if the catalog should be overwritten
+   */
+  private synchronized void loadCatalogIntoCache( String catalogName, final IPentahoSession pentahoSession,
+                                                  boolean overwrite ) {
     ICacheManager cacheMgr = PentahoSystem.getCacheManager( pentahoSession );
     MondrianCatalogCache mondrianCatalogCache = getCacheForRegion( cacheMgr );
-    if ( mondrianCatalogCache.getCatalogs().containsKey( catalogName ) ) {
+    if ( mondrianCatalogCache.getCatalog( catalogName ) != null && !overwrite ) {
```

Comment:
What's the difference between `mondrianCatalogCache.getCatalogs().containsKey( catalogName )` and `mondrianCatalogCache.getCatalog( catalogName ) != null` ?

Reply:
In practice: none. I decided to change to give more visibility at cache api methods instead of relying at a implementation detail.